pinot-core/src/main/java/org/apache/pinot/core/common/datatable/BaseDataTable.java

```diff
@@ -250,6 +250,18 @@ public ByteArray getBytes(int rowId, int colId) {
     return strings;
   }
 
+  @Override
+  public DataTable toMetadataOnlyDataTable() {
+    DataTable destDataTable = DataTableBuilder.getEmptyDataTable();
+    destDataTable.getMetadata().putAll(this.getMetadata());
+    if (!this.getExceptions().isEmpty()) {
```

Comment:
This method needs to be implemented in each individual data table implementation as in V2 the exception is stored in `_metadata`, but in V3 it is stored as a separate map. We don't want to add the exception again if it is already stored in `_metadata`